Tests/Utilities/TestUniq.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2023, Rodrigo Tobar <rtobarc@gmail.com>.
+ * Copyright (c) 2024, Daniel Gaston <tfd@tuta.io>.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/ScopeGuard.h>
+#include <AK/StringView.h>
+#include <LibCore/Command.h>
+#include <LibCore/File.h>
+#include <LibTest/Macros.h>
+#include <LibTest/TestCase.h>
+
+static void run_uniq(Vector<char const*>&& arguments, StringView standard_input, StringView expected_stdout)
+{
+    MUST(arguments.try_insert(0, "uniq"));
+    MUST(arguments.try_append(nullptr));
+    auto uniq = MUST(Core::Command::create("uniq"sv, arguments.data()));
+    MUST(uniq->write(standard_input));
+    auto [stdout, stderr] = MUST(uniq->read_all());
+    auto status = MUST(uniq->status());
+    if (status != Core::Command::ProcessResult::DoneWithZeroExitCode) {
+        FAIL(ByteString::formatted("uniq didn't exit cleanly: status: {}, stdout:{}, stderr: {}", static_cast<int>(status), StringView { stdout.bytes() }, StringView { stderr.bytes() }));
+    }
+    EXPECT_EQ(StringView { expected_stdout.bytes() }, StringView { stdout.bytes() });
+}
+
+TEST_CASE(two_duplicate_lines)
+{
+    run_uniq({}, "AAA\nAAA\n"sv, "AAA\n"sv);
+}
+
+TEST_CASE(two_unique_lines)
+{
+    run_uniq({}, "AAA\nAaA\n"sv, "AAA\nAaA\n"sv);
+}
+
+TEST_CASE(long_line)
+{
+    auto input = Array<u8, 4096> {};
+    auto expected_output = Array<u8, 2048> {};
+    // Create two lines of 2047 A's and a newline.
+    input.fill('A');
+    input[2047] = '\n';
+    input[4095] = '\n';
+
+    expected_output.fill('A');
+    expected_output[2047] = '\n';
+
+    run_uniq({}, StringView { input }, StringView { expected_output });
+}
```

Comment:
I'm not sure if this change is supposed to allow for arbitrarily long lines, so this may be out of scope of this PR, in which case feel free to ignore this. 

Increasing the length of the lines in this test, so that they are greater than 16384 characters causes the test to fail. I believe this happens because the `find_and_populate_until_any_of()` method uses a fixed size circular buffer and the default size of that buffer is 16384.

Reply:
Interesting. In theory, `read_line_with_resize` is supposed to grow the buffer to fit an arbitrarily long line, and the underlying circular buffer will be read into it in chunks if its size is smaller than the line. I will investigate

Reply:
I think this may be an issue with my particular environment, after a full nuke + rebuild, I can't replicate the behavior I was seeing previously. Testing with 64K lines now works fine :confused: 

For reference this is how I modified the test:

```c++
TEST_CASE(long_line)
{
    auto input = Array<u8, 131072> {};
    auto expected_output = Array<u8, 65536> {};
    // Create two lines of 2047 A's and a newline.
    input.fill('A');
    input[65535] = '\n';
    input[131071] = '\n';

    expected_output.fill('A');
    expected_output[65535] = '\n';

    run_uniq({}, StringView { input }, StringView { expected_output });
}
```

Apologies for making you waste your time on this!

Reply:
> What was your test case that failed?

The test case I posted above failed for me prior to me doing a full rebuild. After a full rebuild it works fine, so I don't think there's anything you need to change.

Reply:
> Apologies for making you waste your time on this!
No worries, it was a good thing to test anyway that I overlooked!
> After a full rebuild it works fine, so I don't think there's an issue here.
Good to hear!